### PR TITLE
Donation/Membership form postcode maximum field length validation

### PR DESCRIPTION
**Pull request by Sperling-0**
- Postcode validation allows 16 characters

Ticket: https://phabricator.wikimedia.org/T370617

### CODE REVIEW

**Review by moiikana — APPROVED** [review]

File: src/DataAccess/Migrations/Version20240826150940.php [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+use Doctrine\DBAL\Schema\Schema;
+use Doctrine\Migrations\AbstractMigration;
+
+/**
+ * Auto-generated Migration: Please modify to your needs!
+ */
+final class Version20240826150940 extends AbstractMigration {
+	public function getDescription(): string {
+		return '';
```
Comment (gbirke) [inline src/DataAccess/Migrations/Version20240826150940.php:15]: Please add a description

File: src/DataAccess/Migrations/Version20240826150940.php [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+declare( strict_types=1 );
+
+namespace WMDE\Fundraising\MembershipContext\DataAccess\Migrations;
+
+use Doctrine\DBAL\Schema\Schema;
+use Doctrine\Migrations\AbstractMigration;
+
+/**
+ * Auto-generated Migration: Please modify to your needs!
```
Comment (gbirke) [inline src/DataAccess/Migrations/Version20240826150940.php:11]: Please remove this comment

**Review by gbirke — CHANGES_REQUESTED** [review]

File: src/DataAccess/Migrations/Version20240826150940.php [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+ */
+final class Version20240826150940 extends AbstractMigration {
+	public function getDescription(): string {
+		return '';
+	}
+
+	public function up( Schema $schema ): void {
+		$schema->getTable( 'request' )->getColumn( 'postcode' )->setLength( 16 );
```
Comment (gbirke) [inline src/DataAccess/Migrations/Version20240826150940.php:19]: Sorry, the Migrations use the `real` column names from the database, not the ones from PHP
```suggestion
		$schema->getTable( 'request' )->getColumn( 'plz' )->setLength( 16 );
```

File: src/DataAccess/Migrations/Version20240826150940.php [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+	}
+
+	public function down( Schema $schema ): void {
+		$schema->getTable( 'request' )->getColumn( 'postcode' )->setLength( 8 );
```
Comment (gbirke) [inline src/DataAccess/Migrations/Version20240826150940.php:23]:
```suggestion
		$schema->getTable( 'request' )->getColumn( 'plz' )->setLength( 8 );
```

**Review by gbirke — CHANGES_REQUESTED** [review]

**Review by gbirke — APPROVED** [review]
User-tested, works now!

**Comment by gbirke** [comment]
You can create release v16.2.0 after merging

**Pull request merged**